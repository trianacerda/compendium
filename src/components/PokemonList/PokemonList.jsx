import PropTypes from 'prop-types';
import PokemonDetail from '../PokemonDetail/PokemonDetail';

export default function PokemonList({ pokedex }) {
  return (
    <ul
      aria-label="pokemon-list"
      className="pokemon-List"
      style={{
        backgroundColor: 'grey',
        border: 'solid',
        maxWidth: '50%',
        displayFlex: 'flex',
        justifyContent: 'center',
      }}
    >
      {pokedex.map((pokemon) => {
        return (
          <li className="single-pokemon" key={pokemon.id}>
            <PokemonDetail pokemon={pokemon} />
          </li>
        );
      })}
    </ul>
  );
}

PokemonList.propTypes = {
  pokemon: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      url_image: PropTypes.string.isRequired,
      pokedex: PropTypes.string.isRequired,
      weight: PropTypes.number.isRequired,
      shape: PropTypes.string.isRequired,
      type_1: PropTypes.string.isRequired,
      type_2: PropTypes.string.isRequired,
      speed: PropTypes.number.isRequired,
    }).isRequired
  ),
};
