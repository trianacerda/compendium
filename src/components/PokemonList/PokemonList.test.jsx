import { screen, render } from '@testing-library/react';
import PokemonList from './PokemonList';

it('should render a single pokemon', () => {
  render(
    <PokemonList
      pokedex={[
        {
          id: 11,
          url_image: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/012.png',
          pokedex: 'https://www.pokemon.com/us/pokedex/butterfree',
          weight: 11,
          shape: 'egg',
          type_1: 'bug',
          type_2: 'flying',
          speed: 11,
        },
        {
          id: 24,
          url_image: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/024.png',
          pokedex: 'https://www.pokemon.com/us/pokedex/pidgeotto',
          weight: 100,
          shape: 'upright',
          type_1: 'fairy',
          type_2: 'fire',
          speed: 55,
        },
        {
          id: 81,
          url_image: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/081.png',
          pokedex: 'https://www.pokemon.com/us/pokedex/arbok',
          weight: 10,
          shape: 'egg',
          type_1: 'psychic',
          type_2: 'fire',
          speed: 14,
        },
        {
          id: 20,
          url_image: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/020.png',
          pokedex: 'https://www.pokemon.com/us/pokedex/jigglypuff',
          weight: 20,
          shape: 'egg',
          type_1: 'fairy',
          type_2: 'NA',
          speed: 99,
        },
      ]}
    />
  );

  const pokedex = screen.getByRole('list', { name: 'pokemon-list' });
  expect(pokedex).toMatchSnapshot();
});
