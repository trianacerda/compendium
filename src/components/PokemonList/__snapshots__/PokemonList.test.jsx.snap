// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a single pokemon 1`] = `
<ul
  aria-label="pokemon-list"
  class="pokemon-List"
>
  <li
    class="single-pokemon"
  >
    <figure
      aria-label="pokemon"
    >
      <img
        alt="undefined"
        src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/012.png"
      />
      <div
        class="pokemon-detail"
      >
        <a
          href="https://www.pokemon.com/us/pokedex/butterfree"
        />
        <p>
          Abilities:
          /undefined
        </p>
        <p />
        <p>
          Weight: 
          11
        </p>
        <p>
          Shape: 
          egg
        </p>
        <p>
          Types:
          bug
          /flying
        </p>
        <p>
          Speed:
          11
        </p>
      </div>
    </figure>
  </li>
  <li
    class="single-pokemon"
  >
    <figure
      aria-label="pokemon"
    >
      <img
        alt="undefined"
        src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/024.png"
      />
      <div
        class="pokemon-detail"
      >
        <a
          href="https://www.pokemon.com/us/pokedex/pidgeotto"
        />
        <p>
          Abilities:
          /undefined
        </p>
        <p />
        <p>
          Weight: 
          100
        </p>
        <p>
          Shape: 
          upright
        </p>
        <p>
          Types:
          fairy
          /fire
        </p>
        <p>
          Speed:
          55
        </p>
      </div>
    </figure>
  </li>
  <li
    class="single-pokemon"
  >
    <figure
      aria-label="pokemon"
    >
      <img
        alt="undefined"
        src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/081.png"
      />
      <div
        class="pokemon-detail"
      >
        <a
          href="https://www.pokemon.com/us/pokedex/arbok"
        />
        <p>
          Abilities:
          /undefined
        </p>
        <p />
        <p>
          Weight: 
          10
        </p>
        <p>
          Shape: 
          egg
        </p>
        <p>
          Types:
          psychic
          /fire
        </p>
        <p>
          Speed:
          14
        </p>
      </div>
    </figure>
  </li>
  <li
    class="single-pokemon"
  >
    <figure
      aria-label="pokemon"
    >
      <img
        alt="undefined"
        src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/020.png"
      />
      <div
        class="pokemon-detail"
      >
        <a
          href="https://www.pokemon.com/us/pokedex/jigglypuff"
        />
        <p>
          Abilities:
          /undefined
        </p>
        <p />
        <p>
          Weight: 
          20
        </p>
        <p>
          Shape: 
          egg
        </p>
        <p>
          Types:
          fairy
        </p>
        <p>
          Speed:
          99
        </p>
      </div>
    </figure>
  </li>
</ul>
`;
